File: articles/cell-structure-membranes-organelles/r/membrane_transport.R
# Membrane permeability flux workflow in R.

transport_path <- file.path("data", "membrane_transport_observations.csv")

if (!file.exists(transport_path)) {
  transport_path <- file.path("..", "data", "membrane_transport_observations.csv")
}

transport <- read.csv(transport_path)

transport$flux_concentration_units_um_s <-
  transport$permeability_um_s *
  (transport$external_concentration - transport$internal_concentration)

transport$area_scaled_flux <-
  transport$flux_concentration_units_um_s * transport$membrane_area_um2

print(round(transport, 5))

File: articles/cell-structure-membranes-organelles/python/membrane_transport.py
"""
Membrane transport workflow.

Run:
    python python/membrane_transport.py
"""

from __future__ import annotations

from pathlib import Path

import pandas as pd

from cell_architecture_core import permeability_flux


ARTICLE_DIR = Path(__file__).resolve().parents[1]
TRANSPORT_PATH = ARTICLE_DIR / "data" / "membrane_transport_observations.csv"


def main() -> None:
    """Calculate permeability-limited flux across scenarios."""

    df = pd.read_csv(TRANSPORT_PATH)

    df["flux_concentration_units_um_s"] = [
        permeability_flux(p, cout, cin)
        for p, cout, cin in zip(
            df["permeability_um_s"],
            df["external_concentration"],
            df["internal_concentration"],
        )
    ]

    df["area_scaled_flux"] = df["flux_concentration_units_um_s"] * df["membrane_area_um2"]

    print(df.round(5).to_string(index=False))


if __name__ == "__main__":
    main()
